src/main/kotlin/org/jitsi/nlj/rtcp/KeyframeRequester.kt

```diff
@@ -31,35 +36,94 @@ import org.jitsi.rtp.rtcp.rtcpfb.payload_specific_fb.RtcpFbFirPacketBuilder
  * on what the client supports
  * 3) Aggregation.  This class will pace outgoing requests such that we don't spam the sender
  */
-class KeyframeRequester : ObserverNode("Keyframe Requester") {
+class KeyframeRequester : TransformerNode("Keyframe Requester") {
+
+    private val WAIT_INTERVAL_MS = 100
+
+    override fun transform(packetInfo: PacketInfo): PacketInfo? {
+        val compoundRtcp = packetInfo.packetAs<CompoundRtcpPacket>()
+
+        var drop = false
+        compoundRtcp.packets.forEach { pkt ->
+            when (pkt) {
+                is RtcpFbPacket -> {
```

Comment:
I think this can be simplified a lot. This `when` has only one branch, so if we replace it with an `if` we can avoid one level of indentation.

Reply:
Is this any better now? Anything else we could improve?